docs/user/adding-new-metrics.md

```diff
@@ -6,54 +6,61 @@ The format of that file is documented [here](https://mozilla.github.io/glean_par
 
-category2.subcategory:  # Categories can contain subcategories
-  metric:
-    ...
+For all of these simple value types, you should consider how the value changes in the application and how that relates to the ping it is being sent in. When the ping is submitted, it will contain the "last known" value for the metric, and then the value is reset according to its [lifetime](#What-is-the-lifetime-of-the-data).
 
-```
+If you need to know the current state of a value, such as a user preference, to provide context for other metrics, be aware that the lifetime of the value may not be the same as the lifetime of the ping it is in. You can address this either with a custom ping that matches the lifetime of the value, or by using a user lifetime metric so the value will persist across pings.
 
-The details of the metric parameters are described in [metric parameters](metric-parameters.md).
+### Are you counting things?
 
-The `metrics.yaml` file is used to generate code in the target language (e.g. Kotlin, Swift, ...) that becomes the public API to access your application's metrics.
+If you want to know how many times something happened, use a [counter metric](metrics/counter.html).  If you are counting a group of related things, or you don't know what all of the things to count are at build time, use a [labeled counter metric](metrics/labeled_counter.html).
+
+If you need to know when the things being counted happened relative to other things, use an [event](metrics/event.html).
+
+### Are you measuring time?
+
+If you need to record an absolute time, use a [datetime metric](metrics/datetime.html). Datetimes are recorded in the user's local time, along with a timezone offset from UTC.
+
+If you need to record how long something takes you have a couple of options:
+
+If you need to measure the total time spent doing a particular task, look to the [timespan metric](metric/timespan.html). Note that it can only be used on a single thread.
 
-## Recommendations for defining new metrics
+If you need to measure the relative occurrences of many timings, use a [timing distribution](metric/timing_distribution.html). It builds a histogram of timing measurements, and is safe to use in a multi-threaded context.
 
-“There are only two hard things in Computer Science: cache invalidation and naming things” -- attributed to Phil Karlton.
+If you need to know the time between distinct events that aren't a simple "begin" and "end" pair, consider using an [event](metric/event.html).
+
+### Do you need to know the order of events relative to other events?
+
+If you need to know the order of events relative to other events, such as, the user did A, then B, then C, use an [event metric](metric/event.html).
+
+Events are the most expensive metric type to record, transmit, store and analyse, so they should be used sparingly, and only when one of the other metric types aren't sufficient for answering your question.
```

Comment:
Also perhaps a higher potential for accidental PII, but maybe that's hard to back up with facts.

Reply:
Good point.

Another thing that might be relevant (but that I really struggle to phrase properly) is that events push the computational efforts to the "analysis" step (e.g. think of some of our users using events to compute, at analysis time, the "foreground duration"). Other metric types make sure the effort is at the SDK level, handled by the SDK.

Reply:
It can be backed up with facts: events are structured strings. They contain more bytes per record than any other metric type besides labeled strings, and contain even more bytes overall because a labeled string can only store one string per label: events can record different values each time you hit record.

Put another way, if you were looking for a way to exfiltrate a copy of wikipedia from Fenix to the Data Platform via the Glean SDK, events would be the way you'd choose to do it.